### PR TITLE
PHP 8.4 support added

### PR DESCRIPTION
**Pull request by Atul-glo35265**
|    Q          |   A
|-------------- | ------
| Documentation | yes/no
| Bugfix        | yes/no
| BC Break      | yes/no
| New Feature   | yes/no
| RFC           | yes/no
| QA            | yes/no

### CODE REVIEW

File: .laminas-ci.json [hunk 1]
```diff
@@ -5,7 +5,8 @@
         "intl"
     ],
     "ignore_php_platform_requirements": {
-        "8.3": false
+        "8.3": false,
+        "8.4": false
```
Comment (gsteel) [inline .laminas-ci.json]: 8.4 will need to be `true` here. Very few dependencies will be installable on 8.4

**Review by gsteel — CHANGES_REQUESTED** [review]
The documentation linting problems will need to be fixed prior to merge here - this is due to CI improvements and unrelated to your patch.

Please also sign off your commits as per https://github.com/laminas/.github/blob/main/CONTRIBUTING.md#commit-signoffs

File: .laminas-ci.json (hunk 1 above)
Comment (Atul-glo35265) [reply .laminas-ci.json]: Done

**Comment by Atul-glo35265** [comment]
> The documentation linting problems will need to be fixed prior to merge here - this is due to CI improvements and unrelated to your patch.
> 
> Please also sign off your commits as per https://github.com/laminas/.github/blob/main/CONTRIBUTING.md#commit-signoffs

It seems signoff issue is resolved now.

**Comment by gsteel** [comment]
All implicit nulls need dealing with here. There's also a blocker in `laminas/laminas-stdlib` that will need dealing with.

**Comment by Atul-glo35265** [comment]
> All implicit nulls need dealing with here. There's also a blocker in `laminas/laminas-stdlib` that will need dealing with.

Hi @gsteel, To fix the `laminas/laminas-mvc` related issues [here](https://github.com/laminas/laminas-view/actions/runs/11291736010/job/31406273899), do we need to raise another PR with the fix, for [laminas-mvc:3.7.0](https://github.com/laminas/laminas-mvc/tree/3.7.x) as [3.7.0](https://github.com/laminas/laminas-view/blob/2.36.x/composer.lock#L1812) is being used in the current repo?
Please advise.

**Review by gsteel — CHANGES_REQUESTED** [review]
When you bump a dependency, you must also update the lock file otherwise CI will fail across the board.

**Comment by Atul-glo35265** [comment]
> When you bump a dependency, you must also update the lock file otherwise CI will fail across the board.

I think bumping dependency will not work here, due to other dependencies. Please check [this comment](https://github.com/laminas/laminas-view/pull/253#issuecomment-2410349854) and advise.

**Comment by gsteel** [comment]
Looks like a release of MVC 3.8.0 is required before upgrading to 8.4 here is possible

**Comment by Atul-glo35265** [comment]
> Looks like a release of MVC 3.8.0 is required before upgrading to 8.4 here is possible

@gsteel, can you please confirm, when `laminas/laminas-mvc:3.8.0` release is planned and will 3.8.0 come with php8.4 support? Actually, we are working on Magento Dependencies and php8.4 support. We will be blocked, due to laminas dependencies.

**Comment by gsteel** [comment]
You need to bump the minimum version of PHPUnit 10 to the most recent version in the 10.x series.

I will also need to fix docs linting issues in a different patch.

**Comment by Atul-glo35265** [comment]
> You need to bump the minimum version of PHPUnit 10 to the most recent version in the 10.x series.
> 
> I will also need to fix docs linting issues in a different patch.

I've updated the phpunit10 version to [10.5.36](https://github.com/laminas/laminas-view/pull/253/files#diff-f37acfaa6b11f575a9a6f41a75fa73a61d0f8ebc2c9b8cddc215d8aca10e44f5R3374).

**Comment by gsteel** [comment]
There are still a lot of implicit null params in src and tests here, along with code in dependencies, specifically MVC which you've already fixed in https://github.com/laminas/laminas-mvc/pull/172 - This patch will need to wait for a release of MVC as well as fixing any remaining compatibility issues directly in view src and tests.

**Comment by HeenaBansal20** [comment]
@Atul-glo35265 , When these changes will be merged ?

**Comment by gsteel** [comment]
This has been handled in #255 thanks @Atul-glo35265

**Pull request closed without merging**